docs/src/userguide/language_basics.rst

```diff
@@ -538,8 +538,71 @@ Like other Python looping statements, break and continue may be used in the
 body, and the loop may have an else clause.
 
 
-The include statement
-=====================
+Cython file types
+=================
+
+There are three file types in Cython:
+
+* The implementation files, carrying a ``.pyx`` suffix.
+* The definition files, carrying a ``.pxd`` suffix.
+* The include files, carrying a ``.pxi`` suffix.
+
+The implementation file
+-----------------------
+
+The implementation file, as the name suggest, contains the implementation
+of your functions, classes, extension types, etc. Nearly all the
+python syntax is supported in this file. Most of the time, a ``.py``
+file can be converted into a ``.pyx`` file without changing
+any code, and Cython will retain the python behavior, with a
+slight speed boost.
+
+In addition to the Python syntax, the user can also
+leverage Cython syntax (such as ``cdef``) to use C variables, can
+declare functions as ``cdef`` or ``cpdef`` and can import C definitions
+with :keyword:`cimport`. Many other Cython features usable in implementation files
+can be found throughout this page and the rest of the Cython documentation.
+
+There are some restrictions on the implementation part of an extension type
+if the corresponding definition file also defines that type.
+
+.. note::
+
+    When a ``.pyx`` file is compiled, Cython first checks to see if a corresponding
+    ``.pxd`` file exists and processes it first.
```

Comment:
This would be a good place to explain what the "corresponding .pxd file" is (a .pxd file with the same name as a module), and that it's special compared to other .pxd files in that it contains publicly exported declarations of the implementation file. Users regularly get that wrong and put all sorts of declarations into the .pxd file of a module.